pkg/cloudprovider/vsphere/instances.go

```diff
@@ -165,8 +167,19 @@ func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID s
 		return true, nil
 	}
 
-	klog.V(4).Info("instances.InstanceExistsByProviderID() NOT FOUND with ", uid)
-	return false, err
+	if err != vclib.ErrNoVMFound {
```

Comment:
Should we remove the cache check in line 158?

Reply:
Removed. Should not need to cache.